java/ql/lib/semmle/code/java/PrettyPrintAst.qll

```diff
@@ -385,7 +385,12 @@ private class PpInstanceOfExpr extends PpAst, InstanceOfExpr {
     or
     i = 3 and result = " " and this.isPattern()
     or
-    i = 4 and result = this.getLocalVariableDeclExpr().getName()
+    i = 4 and
+    (
+      result = this.getPattern().asBindingPattern().getName()
+      or
+      result = this.getPattern().asRecordPattern().toString()
```

Comment:
This case doesn't appear to recursively print the entire record pattern, which it really should. The idea of this library is to avoid intermediate string concatenation, so record patterns should be printed using `getChild` and patterns should have their own `PpAst` extension. Let me know if you need more details on this library - it's very useful for inspecting the output of e.g. extractor desugarings.